lib/libzrepl/data_conn.c

```diff
@@ -1213,6 +1245,16 @@ uzfs_zvol_rebuild_scanner(void *arg)
 			    == 1)
 				sleep(5);
 #endif
+
+			snap_zv = uzfs_get_snap_zv_ionum(zinfo,
```

Comment:
calling this multiple times (i.e., for every REBUILD_STEP) may error out without uzfs_close_dataset while sending REBUILD_STEP_DONE (or) we should consider calling this only once for entire volume size

Reply:
need to handle error cases also

Reply:
Fixed partially. Will check error once I take changes from upstream.